### PR TITLE
Tests: Add blake2b Test Cases

### PR DESCRIPTION
**Pull request by 0xFFFC0000**
```
	* New files for blake2b test and testvector.
	* Adding the test to CMakeLists.txt
	* Adding brief documentation for the test.
```

Link to bounty:
https://bounties.monero.social/posts/75/6-500m-blake2b-c-dev-challenge-seraphis

### CODE REVIEW

**Review by selsta — COMMENTED** [review]
Thank you.

I think this should be added to the hash tests, no need for a whole new binary.

Also please add trailing new lines so that GitHub doesn't complain and change the commit title to something like "tests: add blake2b test cases" to be more inline with other commits.

**Comment by 0xFFFC0000** [comment]
Thanks.

Actually I considered adding it into the hash test, but the reason why I went with the new binary is that [bounty](https://bounties.monero.social/posts/75/6-500m-blake2b-c-dev-challenge-seraphis) specifically asks for a new `blake2b_test.cpp` file.

**Comment by 0xFFFC0000** [comment]
Commit message and No trailing newline (editor issue) fixed.

**Comment by 0xFFFC0000** [comment]
I changed the commit and added the test logic to `tests/main.cpp`.

File: tests/crypto/main.cpp
Comment (0xFFFC0000) [reply tests/crypto/main.cpp]: This typo happened when switching to `main.cpp`. Fixed all instances in a new commit.

File: tests/crypto/main.cpp
Comment (0xFFFC0000) [reply tests/crypto/main.cpp]: Fixed. The editor was not showing these on diff.

**Comment by 0xFFFC0000** [comment]
> Looking closer at the distinction between `tests/crypto` and `tests/hash`, I agree with @selsta that `tests/hash` would be more appropriate.

Will change the PR to `tests/hash`.

**Comment by 0xFFFC0000** [comment]
Changed the test location to `tests/hash` folder and integrated it into the existing test suite. 

```
ctest -R hash-blake2b
```

File: tests/README.md
Comment (selsta) [inline tests/README.md:159]: I don't think we need an extra header for blake2b test, they are just another hash test which we already have a header for. You can maybe add some of this documentation to hash test.

File: tests/hash/main.cpp
Comment (selsta) [inline tests/hash/main.cpp:156]: The indentation doesn't match with other parts of the code. You sometimes use 2 spaces and sometimes 4.

File: tests/hash/main.cpp
Comment (selsta) [inline tests/hash/main.cpp:197]: Can you double check all 3 `memset`? The second and third parameter looks wrong.

**Review by selsta — COMMENTED** [review]
A couple small comments.

File: tests/hash/main.cpp
Comment (0xFFFC0000) [reply tests/hash/main.cpp:197]: Nice catch. Although what we have is not required and is not incorrect. 

But semantically, it is incorrect. It only changes first byte. Will fix this.

File: tests/hash/main.cpp
Comment (0xFFFC0000) [reply tests/hash/main.cpp:156]: Editor issue. Will fix this. Have to set up Monero configuration for my editor.

File: tests/README.md
Comment (0xFFFC0000) [reply tests/README.md:159]: Sure. Will fix this.

File: tests/hash/main.cpp
Comment (jeffro256) [inline tests/hash/main.cpp:160]:
```suggestion
      static constexpr size_t HASH_DATA_LEN = 1024;
```

This change would limit this variable to the scope of the `while` loop and won't ever cause unexpected preprocessor tokens to be generated later in the file, especially since this name is so generic.

**Comment by 0xFFFC0000** [comment]
@UkoeHB @jeffro256 @selsta Thanks everyone for reviewing this. This is my first PR to the Monero repository and I am planning to send more PR regularly. 

After @UkoeHB approval, I think at this point I am waiting for final approval.

**Comment by jeffro256** [comment]
> The test data is identical to the [expected vectors](https://github.com/BLAKE2/BLAKE2).

Yup, it's the same file

**Review by jeffro256 — APPROVED** [review]

**Comment by 0xFFFC0000** [comment]
`[ RUN      ] select_outputs.density
/home/runner/work/monero/monero/tests/unit_tests/output_selection.cpp:175: Failure
Expected: (fabsf(selected_ratio - chain_ratio)) < (0.025f), actual: 0.0289887 vs 0.025
[  FAILED  ] select_outputs.density (3142 ms) `

1. Is this related to my PR? 
2. Does this happen in other cases too?

**Comment by 0xFFFC0000** [comment]
@UkoeHB thanks.

**Pull request merged**
